logging: remove tracebacks for UserFacingErrors

orndorffgrant opened the pull request:
## Proposed Commit Message

> logging: remove tracebacks for UserFacingErrors
> 
> UserFacingErrors are used for known situations that
> our code can handle and present the correct message to
> the user. They are not truly "exceptional" and do not
> warrant printing a traceback when they occur. This commit
> changes our logging of these types of errors to continue
> to use the ERROR logging level but no longer include
> tracebacks.
> 
> Fixes: #1586

## Test Steps
<!-- Please include any steps necessary to verify (and reproduce if
this is a bug fix) this change on a live deployed system,
including any necessary configuration files, user-data,
setup, and teardown. Scripts used may be attached directly to this PR. -->

To reproduce:
On a xenial container you'll get tracebacks in logs for several things
```
apt update && apt upgrade
ua detach
# check log for traceback
ua attach # no token
# check log for traceback
ua attach <token>
ua attach <token>
# check log for traceback
ua enable esm-apps
# check log for traceback
```

To see fix, install this version and run through the above steps. You should still see error messages, but no tracebacks.

## Desired commit type::
<!-- put an `x` in one merge type to allow the reviewer to merge for you -->
 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist:
<!-- Go over all the following points, and put an `x` in all the boxes
that apply. -->
 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

blackboxsw reviewed (changes requested):
Good cleanup. Let's also replace logging.exception w/ logging.error on the [KeyboardInterrupt](https://github.com/canonical/ubuntu-advantage-client/blob/main/uaclient/cli.py#L1007)

orndorffgrant commented:
@blackboxsw Just got back around to this. Good catch! I switched that to be a normal `logging.error` with no traceback and adjusted the tests accordingly.

blackboxsw reviewed (approved):
LGTM! Thanks Grant.

blackboxsw reviewed (approved):
LGTM. Filed unrelated "real" integration test failure https://github.com/canonical/ubuntu-advantage-client/issues/1609 now that Xenial is EOL USN/CVE metadata has changed 'needs-triage' status to 'ignored'
